src/Core/src/Fonts/EmbeddedFontLoader.Windows.cs

```diff
@@ -2,50 +2,82 @@
 using System;
 using System.IO;
 using Microsoft.Extensions.Logging;
+using Microsoft.Maui.ApplicationModel;
+using Microsoft.Maui.Storage;
 using Windows.Storage;
 
 namespace Microsoft.Maui
 {
 	/// <inheritdoc/>
 	public partial class EmbeddedFontLoader
 	{
-		const string FontCacheFolderName = "fonts";
+		const string FontCacheFolderName = "Fonts";
 
 		/// <inheritdoc/>
 		public string? LoadFont(EmbeddedFont font)
+		{
+			if (string.IsNullOrWhiteSpace(font.FontName))
+			{
+				throw new InvalidOperationException("FontName for embedded font was null or empty.");
+			}
+			if (font.ResourceStream is null)
+			{
+				throw new InvalidOperationException("ResourceStream for embedded font was null.");
+			}
+
+			return AppInfoUtils.IsPackagedApp
+				? LoadFontPackaged(font.FontName, font.ResourceStream)
+				: LoadFontUnpackaged(font.FontName, font.ResourceStream);
+		}
+
+		private string? LoadFontPackaged(string fontName, Stream resourceStream)
 		{
 			var tmpdir = ApplicationData.Current.LocalFolder.CreateFolderAsync(FontCacheFolderName, CreationCollisionOption.OpenIfExists).AsTask().Result;
 
-			var file = tmpdir.TryGetItemAsync(font.FontName).AsTask().Result;
-			if (file != null)
-				return CleanseFilePath(file.Path);
+			// return an existing file from the cache
+			var existingFile = tmpdir.TryGetItemAsync(fontName).AsTask().Result;
```

Comment:
.Result ? 😱 
Your app called and wanted your code back, but it couldn't because it was deadlocked 😁

Reply:
Been that way since maui 1.0. And yeppy yep, it is the way.

We would have to redesign the whole font system for async. Seems to work, so I ain't touching it.